### ISSUE TITLE
Azure Active Directory Administrator is not updated for existing SQL Server

### ISSUE THREAD

**Issue by l3ender** [Azure/azure-sdk-for-python#22009]
- **Package Name**: azure-mgmt-sql
- **Package Version**: 3.0.1
- **Operating System**: MacOS 11.6
- **Python Version**: 3.9.6

**Describe the bug**
When updating an already-created SQL Server, the Azure Active Directory Administrator parameter is not applied to the server.

**To Reproduce**
The following test Python class shows the issue. It first creates a SQL Server with an administrator (working as expected), and then creates a SQL Server without administrator and tries to update it to add the administrator (not working).

```python
#!/usr/bin/python

import os
from azure.identity._credentials import client_secret
from azure.mgmt.sql import SqlManagementClient
from azure.mgmt.sql.models import Server, ServerExternalAdministrator

resource_group = os.environ.get('AZURE_RESOURCE_GROUP', 'xxx')

subscription_id = os.environ.get('AZURE_SUBSCRIPTION_ID', 'xxx')
credentials = client_secret.ClientSecretCredential(
    client_id=os.environ.get('AZURE_CLIENT_ID', 'xxx'),
    client_secret=os.environ.get('AZURE_CLIENT_SECRET', 'xxx'),
    tenant_id=os.environ.get('AZURE_TENANT_ID', 'xxx'),
)

sql_client = SqlManagementClient(credential=credentials, subscription_id=subscription_id)

# create sql server with AAD admin
params_with_admin = Server(
    location='eastus2',
    version='12.0',
    administrator_login='mylogin',
    administrator_login_password='Password123!',
    administrators=ServerExternalAdministrator(
        administrator_type='ActiveDirectory',
        principal_type='Group',
        login=os.environ.get('AZURE_SQL_AAD_ADMIN_NAME', 'xxx'),
        sid=os.environ.get('AZURE_SQL_AAD_ADMIN_SID', 'xxx'),
        tenant_id=os.environ.get('AZURE_TENANT_ID', 'xxx'),
        azure_ad_only_authentication=False,
    )
)
server_name = 'sql-server-aad-testing-with-admin'
print('creating server {0} with params {1}'.format(server_name, params_with_admin.as_dict()))

response = sql_client.servers.begin_create_or_update(resource_group_name=resource_group,
                                                     server_name=server_name,
                                                     parameters=params_with_admin).result()
print('response: {0}'.format(response.as_dict()))


# create sql server with no initial AAD admin
params_no_admin = Server(
    location='eastus2',
    version='12.0',
    administrator_login='mylogin',
    administrator_login_password='Password123!',
)
server_name = 'sql-server-aad-testing-no-initial-admin'
print('creating server {0} with params {1}'.format(server_name, params_no_admin.as_dict()))

response = sql_client.servers.begin_create_or_update(resource_group_name=resource_group,
                                                     server_name=server_name,
                                                     parameters=params_no_admin).result()
print('response: {0}'.format(response.as_dict()))


# update existing sql server to add AAD admin
print('updating server {0} with params {1}'.format(server_name, params_with_admin.as_dict()))
response = sql_client.servers.begin_create_or_update(resource_group_name=resource_group,
                                                     server_name=server_name,
                                                     parameters=params_with_admin).result()
print('response: {0}'.format(response.as_dict()))
```

**Expected behavior**
Updating an existing SQL Server with Azure Active Directory administrator information should apply the AAD admin change on the server.

**Screenshots**
After the update call is performed, the AAD admin is still not set on the SQL Server:
<img width="744" alt="Screen Shot 2021-12-03 at 10 48 01 PM" src="https://user-images.githubusercontent.com/2243641/144697476-9d750b39-98da-48d4-82d5-46d23f5bbd94.png">

**Additional context**
The following environment variables are used in the script. You can add here or modify the script:

```bash
export AZURE_CLIENT_ID="xxx"
export AZURE_TENANT_ID="xxx"
export AZURE_CLIENT_SECRET="xxx"
export AZURE_SUBSCRIPTION_ID="xxx"
export AZURE_SQL_AAD_ADMIN_NAME="xxx"
export AZURE_SQL_AAD_ADMIN_SID="xxx"
export AZURE_RESOURCE_GROUP="xxx"
```

**Comment by l3ender** [comment]
Hello, any thoughts on this? Is it an issue with the SDK or with the backend API?

**Comment by SaurabhSharma-MSFT** [comment]
@l3ender Thanks for your feedback.  I could see the same behavior in my environment as well and escalating this to service team members to look into this.
![image](https://user-images.githubusercontent.com/38112130/146478150-66eba01d-bf35-4f64-82fa-ed791d490ed5.png)

**Comment by msyyc** [comment]
Hi @service team, please help look into this issue

**Comment by arvindshmicrosoft** [comment]
Sorry for the troubles with this. With a bit of research from the [Azure CLI code](https://github.com/Azure/azure-cli/blob/8fe7dfe67d2f6258d6b8db39c995ff95fb3f36e9/src/azure-cli/azure/cli/command_modules/sql/custom.py#L3682-L3698) which provides equivalent functionality to set the AD admin, a few modifications to the provided code can be done which will allow the functionality to work. I have shared those changed lines below. The main difference is to directly use the [server_azure_ad_only_authentications](https://docs.microsoft.com/en-us/python/api/azure-mgmt-sql/azure.mgmt.sql.operations.serverazureadadministratorsoperations?view=azure-python) class within the SqlManagementClient.

```python
from azure.mgmt.sql.models import Server, ServerExternalAdministrator, AdministratorName
...
# update existing sql server to add AAD admin
aad_admim_params = {
        "principal_type":'Group',
        "login":os.environ.get('AZURE_SQL_AAD_ADMIN_NAME', 'xxx'),
        "sid":os.environ.get('AZURE_SQL_AAD_ADMIN_SID', 'xxx'),
        "tenant_id":os.environ.get('AZURE_TENANT_ID', 'xxx'),
        "administrator_type": 'ActiveDirectory',
        "azure_ad_only_authentication":False,
}

print('updating server {0} with params {1}'.format(server_name, params_with_admin.as_dict()))
response = sql_client.server_azure_ad_administrators.begin_create_or_update(resource_group_name=resource_group,
                                                    administrator_name=AdministratorName.ACTIVE_DIRECTORY,
                                                     server_name=server_name,
                                                     parameters=aad_admin_params).result()
```

**Comment by l3ender** [comment]
Hi @arvindshmicrosoft, thank you for the followup. But I believe you are misunderstanding the issue.

The question is not how to specifically configure a SQL admin, which as you pointed out can be done through the `server_azure_ad_administrators` operations on the SqlManagementClient.

Instead, the problem is that the SQL admin is not correctly configured when using the `servers` operations on the SqlManagementClient. The SQL admin is correctly configured when a server is created, but changes are NOT persisted when wanting to update the admin while making other changes to the server.

In other words, the `sql_client.servers.begin_create_or_update` call does not update the server based on all the parameters.

Please advise. Thank you!

**Comment by arvindshmicrosoft** [comment]
> In other words, the `sql_client.servers.begin_create_or_update` call does not update the server based on all the parameters.

Could you share any other examples where the updates to the server are not persisted when calling this method? Your above repro is very helpful and clearly illustrates the difference in behavior with setting AAD admin after-the-fact. But if there are other facets which when changed via. `servers.begin_create_or_update` are not actually persisted then that is a broader issue.

**Comment by l3ender** [comment]
In my usage to date, I have only noticed the issue with AAD admin not being updated when using `servers.begin_create_or_update`. My apologies for the confusion!

**Comment by arvindshmicrosoft** [comment]
@l3ender - the fact that we are unable to manage the AAD administrator for an existing server using the Servers API (this is not Python SDK specific), is by-design. The SQL AAD team has documented this behavior [here](https://docs.microsoft.com/en-us/azure/azure-sql/database/authentication-azure-ad-only-authentication-create-server?tabs=azure-portal#provision-with-azure-ad-only-authentication-enabled) with the relevant text being:

> To change the existing properties after server or managed instance creation, other existing APIs should be used. For more information, see [Managing Azure AD-only authentication using APIs](https://docs.microsoft.com/en-us/azure/azure-sql/database/authentication-azure-ad-only-authentication#managing-azure-ad-only-authentication-using-apis) and [Configure and manage Azure AD authentication with Azure SQL](https://docs.microsoft.com/en-us/azure/azure-sql/database/authentication-aad-configure).

**Comment by l3ender** [comment]
@arvindshmicrosoft Great detail, and thanks for sharing the doc link! I appreciate the information and will proceed to close this issue as it is working by design. Thanks again!
